trieste/acquisition/function.py

```diff
@@ -1365,3 +1365,29 @@ def repulsion_term(
             )
 
     return acquisition
+
+
+class PredictiveVariance(SingleModelAcquisitionBuilder):
+    """
+    Builder for the determinant of the predictive covariance matrix over the batch points.
+    For a batch of size 1 it is the same as maximizing the predictive variance.
+    """
+
+    def __repr__(self) -> str:
+        """"""
+        return "PredictiveVariance()"
+
+    def prepare_acquisition_function(
+        self, dataset: Dataset, model: ProbabilisticModel
+    ) -> AcquisitionFunction:
+        """
+        :param dataset: Unused.
+        :param model: The model over the specified ``dataset``.
+        :return: The determinant of the predictive function.
+        """
+
+        def determinantcovariance(at: TensorType) -> TensorType:
+            _, covariance = model.predict_joint(at)
+            return tf.linalg.det(covariance)
+
```

Comment:
I wonder if its worth being able to specify an optional jitter param. Also, might be worth returning log(det), as this can be calculated much more numerically stably in tensorflow (https://www.tensorflow.org/api_docs/python/tf/linalg/logdet) .

Reply:
yeah that'll be nice I think. I'll add it.

Reply:
got cholesky decomposition error when using `tf.linalg.logdet` for batch case. I'll investigate why this happens

Reply:
When does the cholesky error pop up? Do you remember to exp afterwards?

Reply:
ah yeah seems i forgot to exp